sale_invoice_plan/models/sale.py

```diff
@@ -0,0 +1,185 @@
+# Copyright 2019 Ecosoft Co., Ltd (http://ecosoft.co.th/)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+from dateutil.relativedelta import relativedelta
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools.float_utils import float_round
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    invoice_plan_ids = fields.One2many(
+        comodel_name="sale.invoice.plan",
+        inverse_name="sale_id",
+        string="Inovice Plan",
+        copy=False,
+    )
+    use_invoice_plan = fields.Boolean(
+        default=False,
+        copy=False,
+    )
+    invoice_plan_process = fields.Boolean(
+        string="Invoice Plan In Process",
+        compute="_compute_invoice_plan_process",
+        help="At least one invoice plan line pending to create invoice",
+    )
+    invoice_plan_total_percent = fields.Float(
+        compute="_compute_invoice_plan_total",
+        string="Percent",
+    )
+    invoice_plan_total_amount = fields.Monetary(
+        compute="_compute_invoice_plan_total",
+        string="Total Amount",
+    )
+    next_installment_date = fields.Date(
+        compute="_compute_next_installment_date",
+        store=True,
+    )
+    invoiced_installment_amount = fields.Monetary(
+        compute="_compute_invoiced_installment_amount",
+        store=True,
+    )
+    pending_installment_amount = fields.Monetary(
+        compute="_compute_pending_installment_amount",
+        store=True,
+    )
+
+    @api.depends("invoice_plan_ids")
+    def _compute_next_installment_date_amount(self):
```

Comment:
```suggestion
    def _compute_next_installment_date(self):
```

Reply:
But somehow, these new methods is not triggered and recomputed. So, currently all the 3 new fields, next_installment_date, invoiced_installment_amount, pending_installment_amount is not computed correctly.